In [1]:
import sunpy

In [3]:
# jupyter nbconvert Jupyter\ Slides.ipynb --to slides --post serve

In [2]:
sunpy.system_info()

SunPy Installation Information

#######
General
#######
Time : Monday, 03. June 2019 06:29PM UT
System : Darwin
Processor : i386
Arch : 64bit
SunPy : 1.0.0
OS: Mac OS X 10.13.6 (i386)


##################
Required Libraries
##################
Python: 3.6.8
NumPy: 1.15.4
SciPy: 1.1.0
matplotlib: 3.0.3
Astropy: 3.1.2
Pandas: 0.23.4
parfive: 0.2.1


#####################
Recommended Libraries
#####################
beautifulsoup: 4.6.3
PyQt4: 4.11.4
PyQt5: NOT INSTALLED
Zeep: 3.3.1
Sqlalchemy: 1.2.12
drms: 0.5.5


# SunPy 1.0

## Key facts
* No more Python 2 support, require Python 3.6+
* Release Schedule
* Existing API is now considered stable (e.g. Map, TimeSeries)
    * Major cleanup and removal of old code (removed many deprecated functions and submodules)
        * TimeSeries instead of LightCurve

See the [what's new](http://docs.sunpy.org/en/stable/whatsnew/1.0.html) for more details!

## Core library improvements
* Coordinates foundation to maps
* Full adoption of Astropy Time
* Fido improvements
* Adoption of a logger system for messaging
* Major improvements in documentation
    * Gallery improvements

## Coordinates and Maps
* Fully featured transformation stack
    * Implemented all major solar-specific coordinate systems
    * Transformations implemented for more general coordinate systems
* WCSAxes provide easy conversion between pixel coordinates and world coordinates for plotting

## SunPy and Time
* Adopted astropy Time, a scientific time object
* 2x64 bit float, maintains sub-nanosecond precision over times spanning the age of the universe.
* Handles leap seconds
* Light travel time corrections
* Behaves similar to python Datetime
    * Provides TimeDelta objects
    * Parses time strings


## Logging System
* No more sending messages to users through print statements!
* All messages provided by SunPy use this logging facility which is based on the Python logging module.
* Messages can have one of several levels
    * **DEBUG**: Detailed information, typically of interest only when diagnosing problems.
    * **INFO**: A message conveying information about the current task, and confirming that things are working as expected
    * **WARNING**: An indication that something unexpected happened, and that user action may be required.
    * **ERROR**: indicates a more serious issue where something failed but the task is continuing
    * **CRITICAL**: A serious error, indicating that the program itself may be unable to continue running.
* Can send all messages (with timestamps) to a file for later reference.
* Configurable through the sunpy configuration system.


## Release Numbering & Schedule
* 2 planned releases of the SunPy core package per year with at least 4 months between them
* Releases are planned for May and November. Follows Astropy releases.
* Two types of releases Long Term Support (LTS) release (1 year support) and a short-support release (6 months).


# Call to Action (i.e. Pledge Drive)
* the SunPy project is not funded and has been developed through the generous contribution of core developers
* All code is developed and 'donated' to the solar physics community
* The survival project depends on the community to contribute to it!
    1. Donate money.
    2. Donate your time.
    3. Help us raise funds (e.g. always cite SunPy in papers and posters, advocate to funding agencies)

## Links and References
* [sunpy.org](sunpy.org)
* Full documentation [docs.sunpy.org](docs.sunpy.org) (hosted on readthedocs, all releases)
* Mailing list [https://groups.google.com/forum/#!forum/sunpy](https://groups.google.com/forum/#!forum/sunpy)
* Weekly open meetings (Wed at noon ET, [https://meet.jit.si/sunpy](https://meet.jit.si/sunpy))
* Chatroom [https://riot.im/app/#/room/#sunpy:openastronomy.org](https://riot.im/app/#/room/#sunpy:openastronomy.org)
